actors/miner/src/lib.rs

```diff
@@ -4771,59 +4764,36 @@ fn check_peer_info(
     Ok(())
 }
 
-fn confirm_sector_proofs_valid_internal(
+// Precondition 1: all sectors referenced have a valid PoRep proof over sealed_cid
+// Precodition 2: all sectors have activated data
```

Comment:
Interesting, what dictates the ordering of Verified allocation -> sector state -> content notification?  Is there any way we can do allocations and content notification together to maintain a single DataActivation call?

Reply:
Getting consistent failure handling and grouping is what dictates it. But I don't think of there being a single data activation concept. If there was, I'd probably use that term to refer to claiming allocations, prior to sector activation, with the notifications to market just that: a notification. But because we need to maintain support for the legacy way, we also need to support the flow of deals before allocations.